idem_azurerm/states/azurerm/web/function_app.py

```diff
@@ -0,0 +1,571 @@
+                return ret
+
+        instrumentation_key = component["instrumentation_key"]
+        # Configures the application insights for the app settings
+        app_settings.append(
+            {"name": "APPINSIGHTS_INSTRUMENTATIONKEY", "value": instrumentation_key}
+        )
+
+    # Builds a storage container named "function-releases" within the specified storage account if not already present
+    container = await hub.exec.azurerm.storage.container.get(
+        ctx,
+        name="function-releases",
+        account=storage_account,
+        resource_group=storage_rg,
+    )
+
+    if "error" in container:
+        container = await hub.exec.azurerm.storage.container.create(
+            ctx,
+            name="function-releases",
+            account=storage_account,
+            resource_group=storage_rg,
+            public_access="None",
+        )
+
+    # Upload the zip file containing the Azure Functions
+    upload_zip = await hub.exec.azurerm.storage.container.upload_blob(
+        ctx,
+        name=filename,
+        container="function-releases",
+        account=storage_account,
+        resource_group=storage_rg,
+        file_path=functions_file_path,
+        overwrite=True,
+    )
+
+    if "error" in upload_zip:
+        log.error(
+            f"Unable to upload {filename} to the function-releases container within the storage account {storage_account}."
+        )
+        ret[
+            "comment"
+        ] = f"Unable to upload {filename} to the function-releases container within the storage account {storage_account}."
+        return ret
+
+    # Retrieves the connection keys for the storage account
+    storage_acct_keys = await hub.exec.azurerm.storage.account.list_keys(
+        ctx, name=storage_account, resource_group=storage_rg
+    )
+    storage_acct_key = storage_acct_keys["keys"][0]["value"]
+
+    # Generate the sas token used within app settings
+    sas_token = generate_account_sas(
+        account_name=storage_account,
+        account_key=storage_acct_key,
+        resource_types=ResourceTypes(object=True, container=True, service=True),
+        permission=AccountSasPermissions(read=True, write=True, list=True, delete=True),
+        expiry=datetime.utcnow() + timedelta(days=2),
+    )
+
+    # Update app settings information from the storage account
+    app_settings.append(
+        {
+            "name": "AzureWebJobsStorage",
+            "value": f"DefaultEndpointsProtocol=https;AccountName={storage_account};AccountKey={storage_acct_key}",
+        }
+    )
+    app_settings.append(
+        {
+            "name": "WEBSITE_RUN_FROM_PACKAGE",
+            "value": f"https://{storage_account}.blob.core.windows.net/function-releases/{filename}?{sas_token}",
+        }
+    )
+
+    # Add any app settings related to a specific OSs
+    if os_type.lower() == "Windows":
```

Comment:
you're setting this to lowercase and matching title case. you'll never hit this condition